packages/vue/src/components/confirm-reset-password.vue

```diff
@@ -0,0 +1,199 @@
+            <base-input
+              name="confirmation_code"
+              required
+              type="number"
+            ></base-input>
+          </base-label>
+
+          <base-label data-amplify-confirmresetpassword-label>
+            <base-text>New password</base-text>
+            <base-input
+              name="password"
+              autocomplete="password"
+              required
+              type="password"
+            ></base-input>
+          </base-label>
+          <base-box data-amplify-lostcode>
+            <slot
+              name="lost-your-code-section"
+              :onLostYourCodeClicked="onLostYourCodeClicked"
+            >
+              <base-text> {{ lostYourCodeText }}</base-text>
+              <base-button type="button" @click.prevent="onLostYourCodeClicked">
+                {{ resendCodeText }}
+              </base-button>
+            </slot>
+          </base-box>
+        </base-field-set>
+
+        <base-footer>
+          <template #footert="{ slotData }">
+            <slot
+              name="footer"
+              :info="slotData"
+              :onBackToSignInClicked="onBackToSignInClicked"
+              :onConfirmResetPasswordSubmit="onConfirmResetPasswordSubmit"
+            >
+            </slot>
+          </template>
+          <base-button type="button" @click.prevent="onBackToSignInClicked">
+            {{ backSignInText }}</base-button
+          >
+          <base-spacer />
+          <base-button
+            :disabled="actorState.matches('confirmResetPassword.pending')"
+            >{{ confirmResetPasswordText }}</base-button
+          >
+        </base-footer>
+        <base-box data-ui-error>
+          {{ actorState?.context?.remoteError }}
+        </base-box>
+      </base-form>
+    </base-wrapper>
+  </slot>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, ComputedRef } from 'vue';
+import BaseHeading from './primitives/base-heading.vue';
+import BaseFieldSet from './primitives/base-field-set.vue';
+import BaseLabel from './primitives/base-label.vue';
+import BaseSpacer from './primitives/base-spacer.vue';
+import BaseButton from './primitives/base-button.vue';
+import BaseFooter from './primitives/base-footer.vue';
+import BaseText from './primitives/base-text.vue';
+import BaseInput from './primitives/base-input.vue';
+import BaseForm from './primitives/base-form.vue';
+import BaseBox from './primitives/base-box.vue';
+import BaseWrapper from './primitives/base-wrapper.vue';
+
+import { useAuth } from '../composables/useAuth';
+
+import {
+  BACK_SIGN_IN_TEXT,
+  CONFIRM_RESET_PASSWORD_TEXT,
+  LOST_YOUR_CODE_TEXT,
+  RESEND_CODE_TEXT,
+  CONFIRM_RESET_PASSWORD_HEADING,
+  CONFIRMATION_CODE_TEXT,
+} from '../defaults/DefaultTexts';
+import {
+  ConfirmResetPasswordSetupReturnTypes,
+  SetupEventContext,
+} from '../types';
+import { getActorState, SignInState } from '@aws-amplify/ui-core';
```

Comment:
Since this is part of `ForgotPassword` flow/actor, we should use `ResetPasswordState` here. My bad for not making this clear.

```suggestion
import { getActorState, ResetPasswordState } from '@aws-amplify/ui-core';
```

Reply:
👍